src/core/api/rtd_csv_transaction/create-sas-token-policy.xml

```diff
@@ -63,18 +63,27 @@
 
             See: https://stackoverflow.com/questions/38474362/get-a-file-sha256-hash-code-and-checksum
         -->
-        <set-variable name="senderCode" value="@(((IResponse)context.Variables["senderCodeResponse"]).Body.As<string>())"/>
 
-        <set-variable name="senderCodeHash" value="@{
+        <choose>
+            <when condition="@(((IResponse)context.Variables["senderCodeResponse"]).StatusCode == 200)" >
+                <set-variable name="senderCode" value="@(((IResponse)context.Variables["senderCodeResponse"]).Body.As<string>())"/>
+            </when>
+            <otherwise>
+                <set-variable name="senderCode" value="unkno"/>
+            </otherwise>
+        </choose>
+
+        <set-variable name="keyHash" value="@{
                 System.Security.Cryptography.SHA256 hasher = System.Security.Cryptography.SHA256.Create();
-                return BitConverter.ToString(hasher.ComputeHash(System.Text.Encoding.UTF8.GetBytes(context.Variables.GetValueOrDefault("senderCode", "")))).Replace("-", "").ToLowerInvariant();
+                    return BitConverter.ToString(hasher.ComputeHash(System.Text.Encoding.UTF8.GetBytes(context.Request.Headers.GetValueOrDefault("Ocp-Apim-Subscription-Key","")))).Replace("-", "").ToLowerInvariant();
```

Comment:
@TommasoLencioni What's the role here of `Replace(...)` ad `ToLowerInvariant(...)`?  As far as I remember, The value of the header `Ocp-Apim-Subscription-key` shouldn't contain `-` or uppercase characters